### PR TITLE
fltk: use a more accurate license identifier

### PR DESCRIPTION
**Pull request by valgur**
Adds FLTK-exceptio: https://spdx.org/licenses/FLTK-exception.html

See https://github.com/fltk/fltk/blob/master/COPYING#L1-L45

### CODE REVIEW

File: recipes/fltk/all/conanfile.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
 class FltkConan(ConanFile):
     name = "fltk"
     description = "Fast Light Toolkit is a cross-platform C++ GUI toolkit"
-    license = "LGPL-2.0-custom"
+    license = "LGPL-2.1+ WITH FLTK-exception"
```
Comment (jwillikers) [inline recipes/fltk/all/conanfile.py:14]:
```suggestion
    license = "LGPL-2.1-or-later WITH FLTK-exception"
```

**Review by jwillikers — CHANGES_REQUESTED** [review]

File: recipes/fltk/all/conanfile.py (hunk 1 above)
Comment (valgur) [reply recipes/fltk/all/conanfile.py:14]: The plus notation is supported by the SPDX expression syntax: https://spdx.github.io/spdx-spec/v2-draft/SPDX-license-expressions/#d3-simple-license-expressions
It's what their online tool spat out for me and I think it's the preferred format thanks to being more generally applicable.
As a data point, it's the format used in the Linux kernel: https://docs.kernel.org/process/license-rules.html

Comment (jwillikers) [reply recipes/fltk/all/conanfile.py:14]: It's deprecated. You have to scroll to the bottom of https://spdx.org/licenses/ to see the deprecated licenses.

Comment (valgur) [reply recipes/fltk/all/conanfile.py:14]: Oh, thanks!

**Review by jwillikers — APPROVED** [review]

**Review by RubenRBS — APPROVED** [review]
Another one! thanks a lot for fixing these :)

**Pull request merged**